### PR TITLE
Row click issue in Category Products tab

### PR DESCRIPTION
**Pull request by ADDISON74**
When you use the **Category Products** tab in the **Categories** page, you may face an important issue without realizing it. As the grid functionality was thought by the Magento team, when you click on a row, regardless of the position where the mouse pointer is, the row will be checked or unchecked. As soon as you press the **[Save Category]** button, products may appear or disappear in the grid by mistake. I faced this issue many times when I was editing the position and I didn't click exactly in the box but a little next to it.

This PR solves the issue and limits the row click strictly to the grid cell where the checkbox is (the first column).

rowclick

### CODE REVIEW

**Comment by ADDISON74** [comment]
@fballiano - I created this PR directly in Github. One commit is coming from a new PR. Please fix this issue for me. Thank you.

**Comment by fballiano** [comment]
which of the files shouldn't be in this PR? I see only these two:

Screenshot 2024-03-03 alle 19 36 57

**Comment by ADDISON74** [comment]
For this one 

/app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml

File: app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml [hunk 1]
```diff
@@ -198,10 +198,10 @@
         grid.reloadParams = {'selected_products[]':categoryProducts.keys()};
     }
     function categoryProductRowClick(grid, event){
-        var trElement = Event.findElement(event, 'tr');
+        var tdElement = Event.findElement(event, 'td');
```
Comment (fballiano) [inline app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml:201]:
```suggestion
        var trElement = Event.findElement(event, 'tr');
```

File: app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml [hunk 2]
```diff
@@ -198,10 +198,10 @@
...
         var isInput   = Event.element(event).tagName == 'INPUT';
-        if(trElement){
-            var checkbox = Element.getElementsBySelector(trElement, 'input');
+        if(tdElement){
+            var checkbox = Element.getElementsBySelector(tdElement, 'input');
```
Comment (fballiano) [inline app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml:204]:
```suggestion
if(trElement){
            var checkbox = Element.getElementsBySelector(trElement, 'input');
```

File: app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml [hunk 3]
```diff
@@ -200,7 +200,7 @@
     function categoryProductRowClick(grid, event){
         var trElement = Event.findElement(event, 'tr');
         var isInput   = Event.element(event).tagName == 'INPUT';
-        if(trElement){
+if(trElement){
```
Comment (fballiano) [inline app/design/adminhtml/default/default/template/catalog/category/edit/form.phtml:203]:
```suggestion
        if(trElement){
```

**Comment by fballiano** [comment]
actually I'd recreate the whole PR not using your "main" branch but another one if possible

**Comment by ADDISON74** [comment]
I close it and create another one. I hope it won't hang any more commits again.

**Pull request closed without merging**